exposee-app/Routes/users.js

```diff
@@ -0,0 +1,86 @@
+import express from 'express';
+import bcrypt from 'bcrypt';
+import { User } from '../models/users.js';
+import { Op } from 'sequelize';
+
+const router = express.Router();
+
+
+router.get ('/user', async(req,res)=> {
+  
+  
+  const users = await User.findAll();
+  console.log(users)
+  res.status(200).json(users);
+  
+     });
+  
+   
+// Route for user registration
+router.post('/user', async (req, res) => {
+  const { username, password, email } = req.body;
+
+  try {
+    // Check if username or email already exists
+    const existingUser = await User.findOne({
+      where: {
+        [Op.or]: [{ username }, { email }]
+      }
+    });
+
+    if (existingUser) {
+      return res.status(400).json({ error: 'Username or email already exists' });
+    }
+
+    // Encrypt the password
+    const hashedPassword = await bcrypt.hash(password, 10);
+
+    // Create a new user
+    const newUser = await User.create({ username, password: hashedPassword, email });
+
+    // Set the user in the session
+    req.session.user = newUser;
+
+    // Return the user data in the response
+    res.json({ user: newUser });
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ error: 'Server error' });
+  }
+});
+
+// Route for user login
+router.post('/users/login', async (req, res) => {
```

Comment:
this would be committed in a future PR, since this is not part of the create-an-account flow